src/main/java/org/kiwiproject/dropwizard/util/job/MonitoredJob.java

```diff
@@ -0,0 +1,95 @@
+package org.kiwiproject.dropwizard.util.job;
+
+import static java.util.Objects.isNull;
+import static java.util.Objects.nonNull;
+import static org.kiwiproject.base.KiwiPreconditions.requireNotBlank;
+import static org.kiwiproject.base.KiwiPreconditions.requireNotNull;
+import static org.kiwiproject.concurrent.Async.doAsync;
+
+import lombok.Builder;
+import lombok.Getter;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.kiwiproject.base.CatchingRunnable;
+import org.kiwiproject.base.DefaultEnvironment;
+import org.kiwiproject.base.KiwiEnvironment;
+
+import java.time.Duration;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Function;
+
+/**
+ * Sets up a job from a {@link Runnable} that can be monitored through health checks to ensure it is running correctly.
+ *
+ * @param <T> a task that extends {@link Runnable}
+ */
+@Slf4j
+@Getter
+public class MonitoredJob<T extends Runnable> implements CatchingRunnable {
```

Comment:
Even though you might see this before some other comments, I'm adding it last, because for some reason it just hit me that there is no good reason this class needs a type parameter, which simplifies _a lot_ of things. If a class implements `Runnable` then it's a `Runnable` and you don't need any type parameter.

I suggest making the other changes first, and then coming back around and removing the type parameter.